internal/information/build_info.go

```diff
@@ -0,0 +1,72 @@
+package information
+
+import (
+	"runtime/debug"
+
+	"github.com/golang/glog"
+)
+
+type BuildInfo struct {
+	commitSha       string
+	architecture    string
+	vcsTime         string
+	vcsType         string
+	operatingSystem string
+	goVersion       string
+}
+
+// GetBuildInfo returns the current Build Information as a BuildInfo type (struct) using a built-in debug package and its ReadBuildInfo() function.
+// Upon error, an empty struct is returned along with a logged error.
+func GetBuildInfo() *BuildInfo {
+	b := new(BuildInfo)
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		glog.Fatal("unable to get build info")
```

Comment:
This is not just logging the error but also terminating program execution.
We should just return an error instead of terminating execution. The decision on what to do when an error happens should be up to the consumers of the type